kubernetes/kube_provider.go

```
@@ -0,0 +1,98 @@
+// usage shares the cache.
+func (p *KubeProvider) ToDiscoveryClient() (discovery.CachedDiscoveryInterface, error) {
+	home, _ := homedir.Dir()
+	httpCacheDir := filepath.Join(home, ".kube", "http-cache")
+	discoveryCacheDir := computeDiscoverCacheDir(filepath.Join(home, ".kube", "cache", "discovery"), p.RestConfig.Host)
+	return diskcached.NewCachedDiscoveryClientForConfig(&p.RestConfig, discoveryCacheDir, httpCacheDir, 10*time.Minute)
+}
+
+// ToRESTMapper returns a deferred-discovery REST mapper wrapped in the
+// kubectl-style shortcut expander.
+func (p *KubeProvider) ToRESTMapper() (meta.RESTMapper, error) {
+	discoveryClient, _ := p.ToDiscoveryClient()
```

golangci-lint (2.12.2)

[error] SA4023(related information): the lhs of the comparison is the 1st return value of this function call

(staticcheck)